<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Variable selection in microbiome compositional data analysis: tutorial</title>
  <meta name="description" content="Methods comparison" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="Variable selection in microbiome compositional data analysis: tutorial" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Methods comparison" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Variable selection in microbiome compositional data analysis: tutorial" />
  
  <meta name="twitter:description" content="Methods comparison" />
  



<meta name="date" content="2020-04-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="selbal.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Variable selection</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#packages-installation-and-loading"><i class="fa fa-check"></i><b>1.1</b> Packages installation and loading</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#example-datasets"><i class="fa fa-check"></i><b>1.2</b> Example datasets</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#crohns-disease"><i class="fa fa-check"></i><b>1.2.1</b> Crohn’s disease</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#high-fat-high-sugar-diet-in-mice"><i class="fa fa-check"></i><b>1.2.2</b> High fat high sugar diet in mice</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="selbal.html"><a href="selbal.html"><i class="fa fa-check"></i><b>2</b> <em>Selbal</em>: selection of balances</a><ul>
<li class="chapter" data-level="2.1" data-path="selbal.html"><a href="selbal.html#crohn-case-study"><i class="fa fa-check"></i><b>2.1</b> Crohn case study</a></li>
<li class="chapter" data-level="2.2" data-path="selbal.html"><a href="selbal.html#hfhs-day1-case-study"><i class="fa fa-check"></i><b>2.2</b> HFHS-Day1 case study</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="clr.html"><a href="clr.html"><i class="fa fa-check"></i><b>3</b> <em>clr-lasso</em></a><ul>
<li class="chapter" data-level="3.1" data-path="clr.html"><a href="clr.html#crohn-case-study-1"><i class="fa fa-check"></i><b>3.1</b> Crohn case study</a></li>
<li class="chapter" data-level="3.2" data-path="clr.html"><a href="clr.html#hfhs-day1-case-study-1"><i class="fa fa-check"></i><b>3.2</b> HFHS-Day1 case study</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="coda.html"><a href="coda.html"><i class="fa fa-check"></i><b>4</b> <em>coda-lasso</em></a><ul>
<li class="chapter" data-level="4.1" data-path="coda.html"><a href="coda.html#crohn-case-study-2"><i class="fa fa-check"></i><b>4.1</b> Crohn case study</a></li>
<li class="chapter" data-level="4.2" data-path="coda.html"><a href="coda.html#hfhs-day1-case-study-2"><i class="fa fa-check"></i><b>4.2</b> HFHS-Day1 case study</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="comparison.html"><a href="comparison.html"><i class="fa fa-check"></i><b>5</b> Concordance of variables selected by the three methods</a><ul>
<li class="chapter" data-level="5.1" data-path="comparison.html"><a href="comparison.html#crohn-case-study-3"><i class="fa fa-check"></i><b>5.1</b> Crohn case study</a><ul>
<li class="chapter" data-level="5.1.1" data-path="comparison.html"><a href="comparison.html#upsetr"><i class="fa fa-check"></i><b>5.1.1</b> UpSetR</a></li>
<li class="chapter" data-level="5.1.2" data-path="comparison.html"><a href="comparison.html#selbal-like-plot"><i class="fa fa-check"></i><b>5.1.2</b> Selbal-like plot</a></li>
<li class="chapter" data-level="5.1.3" data-path="comparison.html"><a href="comparison.html#plotloadings"><i class="fa fa-check"></i><b>5.1.3</b> plotLoadings</a></li>
<li class="chapter" data-level="5.1.4" data-path="comparison.html"><a href="comparison.html#trajectory-plots"><i class="fa fa-check"></i><b>5.1.4</b> Trajectory plots</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="comparison.html"><a href="comparison.html#hfhs-day1-case-study-3"><i class="fa fa-check"></i><b>5.2</b> HFHS-Day1 case study</a><ul>
<li class="chapter" data-level="5.2.1" data-path="comparison.html"><a href="comparison.html#upsetr-1"><i class="fa fa-check"></i><b>5.2.1</b> UpSetR</a></li>
<li class="chapter" data-level="5.2.2" data-path="comparison.html"><a href="comparison.html#selbal-like-plot-1"><i class="fa fa-check"></i><b>5.2.2</b> Selbal-like plot</a></li>
<li class="chapter" data-level="5.2.3" data-path="comparison.html"><a href="comparison.html#plotloadings-1"><i class="fa fa-check"></i><b>5.2.3</b> plotLoadings</a></li>
<li class="chapter" data-level="5.2.4" data-path="comparison.html"><a href="comparison.html#trajectory-plots-1"><i class="fa fa-check"></i><b>5.2.4</b> Trajectory plots</a></li>
<li class="chapter" data-level="5.2.5" data-path="comparison.html"><a href="comparison.html#graphlan"><i class="fa fa-check"></i><b>5.2.5</b> GraPhlAn</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/EvaYiwenWang/Microbiome_variable_selection_tutorial" target="blank">Source codes on github</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Variable selection in microbiome compositional data analysis: tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Variable selection in microbiome compositional data analysis: tutorial</h1>
<p class="author"><em><div style="white-space: pre-line;">Antoni Susin<span class="math inline">\(^1\)</span>, Yiwen Wang<span class="math inline">\(^2\)</span>, Kim-Anh Lê Cao<span class="math inline">\(^2\)</span>, M.Luz Calle<span class="math inline">\(^3\)</span>

<span class="math inline">\(^1\)</span>Mathematical Department, UPC-Barcelona Tech, Barcelona, Spain
<span class="math inline">\(^2\)</span>Melbourne Integrative Genomics, School of Mathematics
and Statistics, The University of Melbourne, Parkville VIC, Australia
<span class="math inline">\(^3\)</span>Biosciences Department, Faculty of Sciences and Technology,
University of Vic - Central University of Catalonia, Vic, Spain</div></em></p>
<p class="date"><em>2020-04-01</em></p>
</div>
<div id="introduction" class="section level1">
<h1><span class="header-section-number">Chapter 1</span> Introduction</h1>
<p>This vignette supports the paper “Variable selection in microbiome compositional data analysis” by Susin <em>et al.</em> (2020) that assesses three compositional data analysis (CoDA) algorithms for microbiome variable selection:</p>
<ul>
<li><em>selbal</em>: a forward selection method for the identification of two groups of taxa whose balance is most associated with the response variable <span class="citation">(Rivera-Pinto et al. <a href="#ref-rivera2018balances">2018</a>)</span>.</li>
<li><em>clr-lasso</em>: penalized regression after the centered log-ratio (clr) transformation <span class="citation">(Zou and Hastie <a href="#ref-zou2005regularization">2005</a>; Tibshirani <a href="#ref-tibshirani1996regression">1996</a>; Le Cessie and Van Houwelingen <a href="#ref-le1992ridge">1992</a>)</span>;</li>
<li><em>coda-lasso</em>: penalized log-contrast regression (log-transformed abundances and a zero-sum constraint on the regression coefficients) <span class="citation">(Lu, Shi, and Li <a href="#ref-lu2019generalized">2019</a>; Lin et al. <a href="#ref-lin2014variable">2014</a>)</span>;</li>
</ul>
<p>Among them, <em>coda-lasso</em> is not yet available as an R package, but the R code for implementing the algorithm is available on Github: <a href="https://github.com/UVic-omics/CoDA-Penalized-Regression" class="uri">https://github.com/UVic-omics/CoDA-Penalized-Regression</a>. Therefore, let us copy the repository first. We only need to copy once, after that, we can update it by fetching the last modified version.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># copy the repository from https://github.com/UVic-omics/CoDA-Penalized-Regression</span>
<span class="co"># system(&#39;git clone https://github.com/UVic-omics/CoDA-Penalized-Regression&#39;)</span>

<span class="co"># fetch the last modified repository from </span>
<span class="co"># https://github.com/UVic-omics/CoDA-Penalized-Regression</span>
<span class="kw">system</span>(<span class="st">&#39;git pull https://github.com/UVic-omics/CoDA-Penalized-Regression&#39;</span>)</code></pre></div>
<p>This vignette only displays the application of all methods on the case studies. Paper related codes and datasets including simulations are all available on GitHub: <a href="https://github.com/UVic-omics/Microbiome-Variable-Selection" class="uri">https://github.com/UVic-omics/Microbiome-Variable-Selection</a></p>
<div id="packages-installation-and-loading" class="section level2">
<h2><span class="header-section-number">1.1</span> Packages installation and loading</h2>
<p>Install then load the following packages:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># cran.packages &lt;- c(&#39;knitr&#39;, &#39;glmnet&#39;, &#39;ggplot2&#39;, &#39;gridExtra&#39;,</span>
<span class="co">#                    &#39;UpSetR&#39;, &#39;ggforce&#39;)</span>
<span class="co"># install.packages(cran.packages)</span>
<span class="co"># devtools::install_github(repo = &#39;UVic-omics/selbal&#39;)</span>

<span class="kw">library</span>(knitr) <span class="co"># rbookdown, kable</span>
<span class="kw">library</span>(glmnet) <span class="co"># glmnet</span>
<span class="kw">library</span>(selbal) <span class="co"># selbal</span>
<span class="kw">library</span>(ggplot2) <span class="co"># draw selbal</span>
<span class="kw">library</span>(gridExtra) <span class="co"># grid.arrange</span>
<span class="kw">library</span>(UpSetR) <span class="co"># upset</span>
<span class="kw">library</span>(ggforce) <span class="co"># selbal-like plot</span>
<span class="kw">library</span>(grid) <span class="co"># grid.draw</span>
<span class="co"># source coda-lasso functions</span>
<span class="kw">source</span>(<span class="dt">file =</span> <span class="st">&#39;./CoDA-Penalized-Regression/R/functions_coda_penalized_regression.R&#39;</span>)

<span class="co"># build in functions</span>
<span class="kw">source</span>(<span class="dt">file =</span> <span class="st">&#39;functions.R&#39;</span>)</code></pre></div>
</div>
<div id="example-datasets" class="section level2">
<h2><span class="header-section-number">1.2</span> Example datasets</h2>
<div id="crohns-disease" class="section level3">
<h3><span class="header-section-number">1.2.1</span> Crohn’s disease</h3>
<p>Crohn’s disease (CD) is an inflammatory bowel disease that has been linked to microbial alterations in the gut. The pediatric CD study <span class="citation">(Gevers et al. <a href="#ref-gevers2014treatment">2014</a>)</span> includes 975 individuals from 662 patients with Crohn’s disease and 313 without any symptoms. The processed data, from 16S rRNA gene sequencing after QIIME 1.7.0, were downloaded from Qiita <span class="citation">(Gonzalez et al. <a href="#ref-gonzalez2018qiita">2018</a>)</span> study ID 1939. The OTU table was agglomerated to the genus level, resulting in a matrix with 48 genera and 975 samples (see Table <a href="index.html#tab:summary">1.1</a>).</p>
<p>Load the data as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">&#39;./datasets/Crohn_data.RData&#39;</span>)</code></pre></div>
<p>File “Crohn_data.RData” contains three objects:</p>
<p><strong>x_Crohn</strong>: the abundance table, a data frame of counts with 975 rows (individuals) and 48 columns (genera)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">class</span>(x_Crohn)</code></pre></div>
<pre><code>## [1] &quot;data.frame&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(x_Crohn)</code></pre></div>
<pre><code>## [1] 975  48</code></pre>
<p><strong>y_Crohn</strong>: a factor variable, indicator of disease status (CD vs. not CD)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">class</span>(y_Crohn)</code></pre></div>
<pre><code>## [1] &quot;factor&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(y_Crohn)</code></pre></div>
<pre><code>##  CD  no 
## 662 313</code></pre>
<p><strong>y_Crohn_numeric</strong>: a numerical variable with values 1 (CD) and 0 (not CD)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">class</span>(y_Crohn_numeric)</code></pre></div>
<pre><code>## [1] &quot;numeric&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">table</span>(y_Crohn_numeric)</code></pre></div>
<pre><code>## y_Crohn_numeric
##   1   2 
## 662 313</code></pre>
<p><em>Note</em>: <strong>x_Crohn</strong> contains no zero. The original matrix of counts (<strong>X</strong>) was transformed by adding one count to each matrix cell: <em>x_Crohn = X + 1</em>. The original matrix of counts can easily be recovered and other imputation methods can be applied.</p>
</div>
<div id="high-fat-high-sugar-diet-in-mice" class="section level3">
<h3><span class="header-section-number">1.2.2</span> High fat high sugar diet in mice</h3>
<p>The study was conducted by Dr Lê Cao at the University of Queensland Diamantina Institute that investigated the effect of diet in mice. C57/B6 female black mice were housed in cages (3 animals per cage and fed with a high fat high sugar diet (HFHS) or a normal diet). Stool sampling was performed at Day 0, 1, 4 and 7. Illumina MiSeq sequencing was used to obtain the 16S rRNA sequencing data. The sequencing data were then processed with QIIME 1.9.0. For our analysis, we considered Day 1 only (HFHSday1 data). The OTU (Operational Taxonomy Units) table after OTU filtering included 558 taxa and 47 samples (24 HFHS diet and 23 normal diet) (see Table <a href="index.html#tab:summary">1.1</a>). Taxonomy information is also available and reported here.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">&#39;./datasets/HFHSday1.RData&#39;</span>)</code></pre></div>
<p>File “HFHSday1.RData” contains three objects:</p>
<p><strong>x_HFHSday1</strong>: the abundance table, a matrix of proportions with 47 rows (samples) and 558 columns (OTUs)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">class</span>(x_HFHSday1)</code></pre></div>
<pre><code>## [1] &quot;matrix&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(x_HFHSday1)</code></pre></div>
<pre><code>## [1]  47 558</code></pre>
<p><strong>y_HFHSday1</strong>: a factor variable, indicator of diet (HFHS vs. normal)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">class</span>(y_HFHSday1)</code></pre></div>
<pre><code>## [1] &quot;factor&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(y_HFHSday1)</code></pre></div>
<pre><code>##   HFHS Normal 
##     24     23</code></pre>
<p><strong>taxonomy_HFHS</strong>: taxonomy table</p>
<p><em>Note</em>: <strong>x_HFHSday1</strong> contains no zero. Zero imputation was performed on the original abundance matrix.</p>
<table>
<caption><span id="tab:summary">Table 1.1: </span> <strong>A summary of the number of samples and number of taxa in each case study</strong></caption>
<thead>
<tr class="header">
<th align="center">Crohn data</th>
<th align="center"></th>
<th align="center">HFHSday1 data</th>
<th align="center"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">No. of genera</td>
<td align="center">48</td>
<td align="center">No. of OTUs</td>
<td align="center">558</td>
</tr>
<tr class="even">
<td align="center">No. of samples</td>
<td align="center">975</td>
<td align="center">No. of samples</td>
<td align="center">47</td>
</tr>
<tr class="odd">
<td align="center">No. of patients with CD</td>
<td align="center">662</td>
<td align="center">No. of mice with HFHS diet</td>
<td align="center">24</td>
</tr>
<tr class="even">
<td align="center">No. of healthy patients</td>
<td align="center">313</td>
<td align="center">No. of mice with normal diet</td>
<td align="center">23</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-gevers2014treatment">
<p>Gevers, Dirk, Subra Kugathasan, Lee A Denson, Yoshiki Vázquez-Baeza, Will Van Treuren, Boyu Ren, Emma Schwager, et al. 2014. “The Treatment-Naive Microbiome in New-Onset Crohn’s Disease.” <em>Cell Host &amp; Microbe</em> 15 (3). Elsevier: 382–92.</p>
</div>
<div id="ref-gonzalez2018qiita">
<p>Gonzalez, Antonio, Jose A Navas-Molina, Tomasz Kosciolek, Daniel McDonald, Yoshiki Vázquez-Baeza, Gail Ackermann, Jeff DeReus, et al. 2018. “Qiita: Rapid, Web-Enabled Microbiome Meta-Analysis.” <em>Nature Methods</em> 15 (10). Nature Publishing Group: 796–98.</p>
</div>
<div id="ref-le1992ridge">
<p>Le Cessie, Saskia, and Johannes C Van Houwelingen. 1992. “Ridge Estimators in Logistic Regression.” <em>Journal of the Royal Statistical Society: Series C (Applied Statistics)</em> 41 (1). Wiley Online Library: 191–201.</p>
</div>
<div id="ref-lin2014variable">
<p>Lin, Wei, Pixu Shi, Rui Feng, and Hongzhe Li. 2014. “Variable Selection in Regression with Compositional Covariates.” <em>Biometrika</em> 101 (4). Oxford University Press: 785–97.</p>
</div>
<div id="ref-lu2019generalized">
<p>Lu, Jiarui, Pixu Shi, and Hongzhe Li. 2019. “Generalized Linear Models with Linear Constraints for Microbiome Compositional Data.” <em>Biometrics</em> 75 (1). Wiley Online Library: 235–44.</p>
</div>
<div id="ref-rivera2018balances">
<p>Rivera-Pinto, J, JJ Egozcue, Vera Pawlowsky-Glahn, Raul Paredes, Marc Noguera-Julian, and ML Calle. 2018. “Balances: A New Perspective for Microbiome Analysis.” <em>MSystems</em> 3 (4). Am Soc Microbiol: e00053–18.</p>
</div>
<div id="ref-tibshirani1996regression">
<p>Tibshirani, Robert. 1996. “Regression Shrinkage and Selection via the Lasso.” <em>Journal of the Royal Statistical Society: Series B (Methodological)</em> 58 (1). Wiley Online Library: 267–88.</p>
</div>
<div id="ref-zou2005regularization">
<p>Zou, Hui, and Trevor Hastie. 2005. “Regularization and Variable Selection via the Elastic Net.” <em>Journal of the Royal Statistical Society: Series B (Statistical Methodology)</em> 67 (2). Wiley Online Library: 301–20.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="selbal.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Microbiome_variable_selection_tutorial.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
